dep_check/use_cases/check.py

```diff
@@ -122,4 +126,5 @@ def run(self) -> ExitCode:
         unused = self.all_rules.difference(self.used_rules)
         self.report_printer.print_report(errors, unused, nb_files)
 
-        return ExitCode.OK if not errors else ExitCode.KO
+        if errors:
+            raise ForbiddenDepencyError
```

Comment:
Pass `errors` as an argument to the exception constructor ? Lets the caller inspect it etc.

Reply:
I'm not sure this is useful but I can add it in the exception

Reply:
You should not if there is no clearly identified value, it could still be added later.

Reply:
agreed, printing the report is part of the use case, I removed it for now.